api/storage_service.cc

```diff
@@ -1766,6 +1766,21 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
         co_return json::json_return_type(static_cast<int>(scrub_status::successful));
     });
 
+    ss::start_backup.set(r, [&snap_ctl] (std::unique_ptr<http::request> req) -> future<json::json_return_type> {
+        auto endpoint = req->get_query_param("endpoint");
+        auto keyspace = req->get_query_param("keyspace");
+        auto bucket = req->get_query_param("bucket");
+        auto snapshot_name = req->get_query_param("snapshot");
+        if (snapshot_name.empty()) {
+            // TODO: If missing, snapshot should be taken by scylla, then removed
+            throw httpd::bad_param_exception("The snapshot name must be specified");
+        }
+
+        auto& ctl = snap_ctl.local();
+        auto task_id = co_await ctl.start_backup(std::move(endpoint), std::move(bucket), std::move(keyspace), std::move(snapshot_name));
+        co_return json::json_return_type(fmt::to_string(task_id));
```

Comment:
We should have a combined snapshot-into-name + backup + drop snapshot API, but that can come later.

Reply:
I'm not big fan of do-everything-in-one-API-call approach. I still think that preparing a snapshot is better be done by the caller, but have no strong enough arguments for that, so yes -- will probably come later

Reply:
yeah, i concur with Pavel on this. would be simpler if we can provide a set of lower-level APIs instead of this combo API?